### DISCUSSION TITLE
gatsby-source-contentful - query for content / assets by Contentful tag

### DISCUSSION THREAD

**Discussion by GregorShear** [gatsbyjs/gatsby#31000] [Help] [answered]
Contentful's content and asset models have a `contentfulMetadata.tags` array attribute, but I'm not seeing this exposed through the Gatsby GraphQL interface. I'd love to be able to query for image assets that match a particular tag - is this on the roadmap?

**Comment by LekoArts** [comment, marked as answer]
cc @axe312ger

**Reply by axe312ger** [reply]
Yes it is. We plan to implement it in the next major.

Currently working on #30855 which will allow us to add this new Contentful feature

**Reply by rtuttlecoder** [reply]
any ETA on this aspect?
Is there another way to query content_type by tag in the meantime?

**Reply by axe312ger** [reply]
@rtuttlecoder tag support is enabled since 5.10.0 (2021-07-20).

**Reply by GregorShear** [reply]
@axe312ger Indeed - I'm able to query my content objects by `metadata.tag` now, but media assets are still missing that `metadata` attribute. Is this expected?

**Reply by axe312ger** [reply]
@GregorShear asset tag support was implemented in January: https://github.com/gatsbyjs/gatsby/pull/34480

Your source plugin is probably too old.
